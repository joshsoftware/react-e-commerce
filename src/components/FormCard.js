import React from 'react';
import RegistrationForm from './RegistrationForm';
import LoginForm from './LoginForm';
import PropTypes from 'prop-types';
import FormCardComponent from './FormCardComponent';
import CardBodyWrapper from './CardBodyWrapper';

const FormCard = ({ type, validateData, dispatch, formState }) => {
  if (type === 'registration') {
    return (
      <FormCardComponent
        data={
          <CardBodyWrapper
            body_content={
              <RegistrationForm
                validateData={validateData}
                dispatch={dispatch}
                formState={formState}
              />
            }
          />
        }
      />
    );
  } else {
    return (
      <FormCardComponent
        data={
          <CardBodyWrapper
            body_content={
              <LoginForm validateData={validateData} dispatch={dispatch} formState={formState} />
            }
          />
        }
      />
    );
  }
};

export default FormCard;

FormCard.propTypes = {
  type: PropTypes.string.isRequired
};
