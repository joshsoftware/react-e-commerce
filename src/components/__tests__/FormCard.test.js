import { render, cleanup } from '@testing-library/react';
import React from 'react';
import FormCard from '../FormCard';

afterEach(cleanup);

describe('FormCard Component', () => {
  it('Must render RegistrationForm Component', () => {
    const { asFragment } = render(<FormCard type="registration" />);
    expect(asFragment()).toMatchSnapshot();
  });
});
