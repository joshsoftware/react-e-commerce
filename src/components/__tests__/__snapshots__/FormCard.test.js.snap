// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormCard Component Must render RegistrationForm Component 1`] = `
<DocumentFragment>
  <div
    class="card"
  >
    <div
      class="card-body"
    >
      <h3>
        Register
      </h3>
      <hr />
      <form
        class=""
      >
        <div
          class="form-group"
        >
          <label
            class=""
            for="exampleEmail"
          >
            Email
          </label>
          <input
            aria-invalid="true"
            class="is-invalid form-control"
            name="email"
            placeholder="example@company.com"
            type="email"
            value=""
          />
          <div
            class="invalid-feedback"
          />
        </div>
        <div
          class="form-group"
        >
          <label
            class=""
            for="examplePassword"
          >
            Password
          </label>
          <input
            aria-invalid="true"
            class="is-invalid form-control"
            name="password"
            placeholder="password"
            type="password"
            value=""
          />
          <div
            class="invalid-feedback"
          />
        </div>
        <div
          class="form-group"
        >
          <label
            class=""
            for="exampleFirstName"
          >
            First Name
          </label>
          <input
            aria-invalid="true"
            class="is-invalid form-control"
            name="firstname"
            placeholder="John"
            type="firstname"
            value=""
          />
          <div
            class="invalid-feedback"
          />
        </div>
        <div
          class="form-group"
        >
          <label
            class=""
            for="exampleLastName"
          >
            Last Name
          </label>
          <input
            aria-invalid="true"
            class="is-invalid form-control"
            name="lastname"
            placeholder="Doe"
            type="lastname"
            value=""
          />
          <div
            class="invalid-feedback"
          />
        </div>
        <div
          class="form-group"
        >
          <label
            class=""
            for="exampleCountry"
          >
            Country
          </label>
          <input
            aria-invalid="true"
            class="is-invalid form-control"
            name="country"
            placeholder=""
            type="country"
            value=""
          />
          <div
            class="invalid-feedback"
          />
        </div>
        <div
          class="form-group"
        >
          <label
            class=""
            for="exampleState"
          >
            State
          </label>
          <input
            aria-invalid="true"
            class="is-invalid form-control"
            name="state"
            placeholder=""
            type="state"
            value=""
          />
          <div
            class="invalid-feedback"
          />
        </div>
        <div
          class="form-group"
        >
          <label
            class=""
            for="exampleCity"
          >
            City
          </label>
          <input
            aria-invalid="true"
            class="is-invalid form-control"
            name="city"
            placeholder=""
            type="city"
            value=""
          />
          <div
            class="invalid-feedback"
          />
        </div>
        <div
          class="form-group"
        >
          <label
            class=""
            for="exampleAddress"
          >
            Address
          </label>
          <input
            aria-invalid="true"
            class="is-invalid form-control"
            name="address"
            placeholder=""
            type="address"
            value=""
          />
          <div
            class="invalid-feedback"
          />
        </div>
        <button
          class="btn btn-secondary"
          type="button"
        >
          Submit
        </button>
      </form>
      <h6>
        <br />
        Already registered? login
      </h6>
    </div>
  </div>
</DocumentFragment>
`;
